### PR TITLE
Fixed build failure and Add some changes in code

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 1]
```diff
@@ -228,71 +250,120 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
...
             connection = (HttpsURLConnection) smsProviderUrl.openConnection();
             connection.setDoInput(true);
             connection.setDoOutput(true);
-            connection.setRequestMethod(SMSOTPConstants.HTTP_METHOD);
-            if (connection.getResponseCode() == 200) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Code is successfully sent to your mobile number");
+            String[] headerList;
+            if (!headerString.isEmpty()) {
+                headerString = headerString.trim().replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
```
Comment (hmrajas) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Use constant for "\\$ctx.num".

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 2]
```diff
@@ -228,71 +250,120 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
...
 
-    public boolean sendRESTCall(String url, String urlParameters) throws IOException {
+    public boolean sendRESTCall(String smsUrl, String httpMethod, String headerString, String payload,
+                                String httpResponse) throws IOException, AuthenticationFailedException {
+
         HttpsURLConnection connection = null;
+        String smsMessage = SMSOTPConstants.SMS_MESSAGE;
         try {
-            URL smsProviderUrl = new URL(url + urlParameters);
+            smsUrl = smsUrl.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
```
Comment (hmrajas) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Use constant for "\\$ctx.msg".

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 3]
```diff
@@ -228,71 +250,120 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
...
+            if (httpMethod.toUpperCase().equals(SMSOTPConstants.GET_METHOD)) {
+                connection.setRequestMethod(SMSOTPConstants.GET_METHOD);
+            } else if (httpMethod.toUpperCase().equals(SMSOTPConstants.POST_METHOD)) {
+                connection.setRequestMethod(SMSOTPConstants.POST_METHOD);
+                if (!payload.isEmpty()) {
+                    payload = payload.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg", smsMessage + otpToken);
+                }
+                OutputStreamWriter writer = null;
+                try {
+                    writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8");
```
Comment (hmrajas) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Use constant for "UTF-8".

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 4]
```diff
@@ -192,14 +148,80 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
...
         String contextToken = (String) authContext.getProperty(otpToken);
         if (userToken.equals(contextToken)) {
-            context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier("an authorised user"));
+            context.setSubject(AuthenticatedUser
+                    .createLocalAuthenticatedUserFromSubjectIdentifier("an authorised user"));
         } else {
-            log.error("Code Mismatch");
-            throw new AuthenticationFailedException("Code mismatch");
+            String username = getUsername(context);
+            if (username != null) {
```
Comment (keerthu) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Use StringUtils.isEmpty() for null check. Fix this throughout the PR.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 2 above)
Comment (Kanapriya) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in 1e74de47c459cfc5bb982f0e2713d62e87da0795

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 1 above)
Comment (Kanapriya) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in 1e74de47c459cfc5bb982f0e2713d62e87da0795

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 3 above)
Comment (Kanapriya) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in 1e74de47c459cfc5bb982f0e2713d62e87da0795

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 4 above)
Comment (Kanapriya) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in 1e74de47c459cfc5bb982f0e2713d62e87da0795

**Pull request merged**